src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+
+        var calendarData : Map<string, SwellingDay> = new Map<string, SwellingDay>();
+        swellingEntries.forEach( swellingEntry => {
+            let key : string = startOfDay(parseISO(swellingEntry.dateRecorded)).toDateString();
+            let exists = calendarData.get(key);
+            if (exists){
+                if (parseISO(swellingEntry.dateRecorded) > parseISO(exists.dateRecorded)){
+                    exists.severity = swellingEntry.severity;
+                    calendarData.set(key, exists);
+                }
+            }
+            else {
+                calendarData.set(key, swellingEntry);
+            }
+        });
+
+        setData(calendarData);
+    }
+
+    function createResultsMap(results : SurveyAnswer[]){
+        var groupedByResult = new  Map<Guid, SwellingDay>();
+
+        results.forEach( (r) => {
+            var exists = groupedByResult.get(r.surveyResultID);
+            var swellingDay = exists ?? {dateRecorded : "", severity : ""};
+
+            if (r.resultIdentifier === props.dateRecordedResultIdentifier && r.answers)
+            {
+                swellingDay.dateRecorded = r.answers[0];
+            }
+
+            if (r.resultIdentifier === props.severityResultIdentifier && r.answers)
+            {
+                var severity = "";
+                switch (r.answers[0]){
+                    case "mild":
+                    case "severity1":
+                        severity = "mild";
+                        break;
+                    case "moderate":
+                    case "severity2":
+                        severity = "moderate";
+                        break;
+                    case "severe":
+                    case "severity3":
+                        severity = "severe";
+                        break;
+                    }
+
+                    swellingDay.severity = severity;
+            }
+
+            groupedByResult.set(r.surveyResultID, swellingDay);
+        });
+
+        return groupedByResult;
+    }
+
+    const renderDay = (year: number, month: number, day?: number): React.JSX.Element => {
+        return <CalendarDay
+            year={year}
+            month={month}
+            day={day}
+            stateConfiguration={stateConfiguration}
+            computeStateForDay={computeStateForDay}
+        />;
+    };
+
+    function pageMonth(newStart: Date) {
+        setMonth(startOfMonth(newStart));
+    } 
+
+
+    useEffect(() => {
+        initialize();
+        MyDataHelps.on("applicationDidBecomeVisible", initialize);
+
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", initialize);
+        }
+    }, []);
+
+
+    if (!data){
+        return <LoadingIndicator/>
+    }
+    else {
+        return <>
+            <DateRangeNavigator intervalType={"Month"} intervalStart={datePagerMonth} onIntervalChange={pageMonth}></DateRangeNavigator>
```

Comment:
The `DateRangeNavigator` should not be part of this component.  The `DateRangeNavigator` will already be provided by a wrapping `DateRangeCoordinator`.  You'll just want to reference a `DateRangeContext` in this component to get the correct dates to use.

```
const dateRangeContext = useContext(DateRangeContext);
```


Here is the Asthma Control Calendar (not in PR yet) to peek at for an example:  https://github.com/CareEvolution/MyDataHelpsUI/blob/AsthmaUI/src/components/asthma/components/AsthmaControlCalendar/AsthmaControlCalendar.tsx

You can also check out the Symptom Shark calendar for another example: https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/components/symptom-shark/presentational/Calendar/Calendar.tsx